packages/js-moi-providers/src.ts/abstract-provider.ts

```diff
@@ -21,14 +21,31 @@ import {
     Registry,
     Status,
     SyncStatus,
-    TDU
+    TDU,
+    type Log,
+    type LogFilter
 } from "../types/jsonrpc";
 
+export type NewTesseractsByAccount = { event: "newTesseractsByAccount", params: string };
+
+export type NewLogs = { event: "newLogs", params: LogFilter };
+
+export type ProviderEvents = keyof WebsocketEventMap | { event: string, params: any };
+
+export interface WebsocketEventMap {
+    newTesseracts: [tesseract: Tesseract];
+    newPendingInteractions: [interactionHash: string];
+    connect: [];
+    error: [error: unknown];
+    close: [];
+    reconnect: [attempt: number];
+}
```

Comment:
Same here maintain the types separately.

Reply:
https://github.com/sarvalabs/js-moi-sdk/pull/88/commits/c8e7b7e57d09668b41fbd4ba912a09991969e39e